projects/ion/src/lib/chip/chip.component.ts

```diff
@@ -96,4 +115,45 @@ export class ChipComponent {
   dropdownSearchChange(value: string): void {
     this.dropdownSearchEvents.emit(value);
   }
+
+  checkTargetClick = (event): void => {
+    if (event.target instanceof EventTarget) {
+      const target = event.target as HTMLInputElement;
+      const elementChip = target.closest('#' + this.id);
+      const elementChipDropdown = target.closest('#dropdown-' + this.id);
+      const elementChipSearch = target.closest(
+        '#dropdown-' + this.id + ' .dropdown-search'
+      );
+
+      if (elementChipSearch || (elementChipDropdown && this.multiple)) {
+        return;
+      }
+
+      if (!elementChip) {
+        if ((this.previusSelectedStatus && this.selected) || !this.selected) {
+          this.showDropdown = false;
+          this.selected = false;
+          this.events.emit({
+            selected: this.selected,
+            disabled: this.disabled,
+          });
+          this.ref.detectChanges();
+        }
+      }
+      this.previusSelectedStatus = this.selected;
+    }
+  };
+
+  ngAfterViewInit(): void {
+    setTimeout(() => {
+      this.id = generateIDs('chip-', 'ion-chip');
+      this.ref.markForCheck();
+    }, COOLDOWN_TIME);
+
+    document.body.addEventListener('click', this.checkTargetClick);
```

Comment:
As mentioned just below:

```suggestion
    document.body.addEventListener('click', (event) => {
      this.checkTargetClick(event);
    });
```

Reply:
Another alternative that does not affect coverage and would not require the use of ngOnDestroy, is to add the once property to addEventListener, which guarantees that it is called once and is automatically removed:
```suggestion
    document.body.addEventListener('click', this.checkTargetClick, {
      once: true,
    });
```